transport/middleware.go

```diff
@@ -1,19 +1,80 @@
 package transport
 
 import (
-	"fmt"
+	"errors"
 	"net/http"
+	"strings"
+
+	reperr "release-manager/repository/errors"
+	neterr "release-manager/transport/errors"
+	"release-manager/transport/model"
+
+	httpx "go.strv.io/net/http"
 )
 
-func (h *Handler) recoverPanic(next http.Handler) http.Handler {
+func (h *Handler) auth(next http.Handler) http.Handler {
 	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-		defer func() {
-			if err := recover(); err != nil {
-				w.Header().Set("Connection", "close")
-				WriteServerErrorResponse(w, r, fmt.Errorf("%s", err))
+
+		w.Header().Add("Vary", "Authorization")
```

Comment:
I believe you can get rid of Vary, browsers don't really support it and cache control happens based on Cache-Control or status response.